builtin/game/chat.lua

```diff
@@ -305,23 +312,69 @@ local function handle_revoke_command(caller, revokename, revokeprivstr)
 		end
 	end
 
+	local privs_unknown = ""
+	local basic_privs =
+		core.string_to_privs(core.settings:get("basic_privs") or "interact,shout")
+	local irrevokable = {}
+	local has_irrevokable_priv = false
+	for priv, _ in pairs(revokeprivs) do
+		if not basic_privs[priv] and not caller_privs.privs then
+			return false, S("Your privileges are insufficient "..
+					"(basic_privs only allows you to revoke: @1).",
+					core.privs_to_string(basic_privs, ', '))
+		end
+		local def = core.registered_privileges[priv]
+		if not def then
+			privs_unknown = privs_unknown .. S("Unknown privilege: @1", priv) .. "\n"
+		elseif is_singleplayer and def.give_to_singleplayer then
+			irrevokable[priv] = true
+			has_irrevokable_priv = true
+		elseif is_admin and def.give_to_admin then
+			irrevokable[priv] = true
+			has_irrevokable_priv = true
```

Comment:
This could be set inside the `for .. (irrevokable= ` loop below to ensure it's exactly set when needed, and not in each check.